security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -114,6 +121,54 @@ func (sf *SecretFetcher) Run(ch chan struct{}) {
 	cache.WaitForCacheSync(ch, sf.scrtController.HasSynced)
 }
 
+// Generate implements the XDS Generate method for SDS.
+// We also allow returning the secrets over the main XDS connection,
+// for efficiency.
+// This method only allows returning secret for authenticated requests.
+func (sf *SecretFetcher) Generate(proxy *istiomodel.Proxy, push *istiomodel.PushContext,
+	w *istiomodel.WatchedResource, updates istiomodel.XdsUpdates) istiomodel.Resources {
+	// 1. Get the namespace of the caller
+	// TODO: add an error to Generate (for non-empty)
+
+	// This depends on in-progress PR to VERIFY the namespace.
+
+	// 2. If istio-system - ok, can read secrets in istio-system
```

Comment:
this comment is inconsistent with the implementation below which simply enforces that secrets should be in the same namespace as the proxy.

Reply:
Good catch, the comment is right - will fix the implemnentation.